soccer/gameplay/tactics/positions/celebration.py

```diff
@@ -0,0 +1,77 @@
+import behavior
+import robocup
+import constants
+import single_robot_composite_behavior
+import main
+import enum
+import skills
+import random
+import time
+
+class Celebration(single_robot_composite_behavior.SingleRobotCompositeBehavior):
+	MaxSpinAngle = 360
+	SpinPerTick = 1
+
+	class State(enum.Enum) :
+		run_around = 0
+
+		spin = 1
+
+	def __init__(self):
+		super().__init__(continuous=True)
+
+		for s in Celebration.State:
+			self.add_state(s, behavior.Behavior.State.running)
+
+		self.add_transition(behavior.Behavior.State.start, Celebration.State.run_around,
+							lambda: True, 'immediately')
+
+		self.add_transition(Celebration.State.run_around, Celebration.State.spin,
+							lambda: self.spin_time,
+                				 'time to yeet')
+
+		self.add_transition(Celebration.State.spin, Celebration.State.run_around,
+							lambda: self.im_dizzy(),
+								'time to go running')
+
+		self.spin_angle = 0
+		self.spin_time = False
+		r = constants.Robot.Radius
+		self.corners = [robocup.Point(-constants.Field.Width/2 + r, r), robocup.Point(constants.Field.Width/2 - r, r),
+						robocup.Point(constants.Field.Width/2 - r, constants.Field.Length - r),
+						robocup.Point(-constants.Field.Width/2 + r, constants.Field.Length - r),
+						robocup.Point(0, constants.Field.Length/2)]
+
+		self.current_corner = 0
+		self.start_time = time.time()
+
+
+	def on_enter_run_around(self):
+		self.current_corner = random.randint(0,4)
+		self.robot.move_to_direct(self.corners[0])
```

Comment:
This feels wrong to force to corner zero even though you have a variable called current corner. It doesn't make a difference since running is called the same frame, but you shouldn't rely on that